hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/impl/ITestXAttrCost.java

```diff
@@ -64,6 +64,9 @@ public ITestXAttrCost() {
   @Test
   public void testXAttrRoot() throws Throwable {
     describe("Test xattr on root");
+    // TODO: Previously a call to getObjectMetadata for a base path, ie with an empty key would
+    //  return some metadata. (bucket region, content type). headObject() fails without a key, check
+    // how this can be fixed. 
     Path root = new Path("/");
```

Comment:
The issue is that with the V1 Client, when you did `s3V1.getObjectMetadata("my-bucket", "") `, so the key was empty as you were probing the root path, this worked ok and returned the bucket region and content-type. With V2, when you do `s3v2.headObject(HeadObjectRequest.builder().bucket("my-bucket").build())`, you get an error as the key must be specified. Instead we’ll need to call headBucket() here, but this has other implications. We will need some special code to handle this. Do we need to support XAttr operations on root dirs?

The other case is in testDelegatedFileSystem, where getObjectMetadata is used as a probe to check endpoint correctness. Can this simply be replaced by headBucket()?